Wip messaging

jakubwitowski-silvair opened the pull request.

adamgralec-silvair committed:
Initial version of send message API.

jakubwitowski-silvair committed:
Unicast address restoring fixed

adamgralec-silvair committed:
Added DBus variant to choose between using direct messages or group messages while sending.

jakubwitowski-silvair committed:
Write null to Json (ividx, ivupdate and ttl) when node is unprovisioned

jakubwitowski-silvair committed:
MessageReceived signal emit implemented

jakubwitowski-silvair committed:
Remove redundant function a use dbus.c api

adamgralec-silvair committed:
Added real netkey proprty on Node interface.

adamgralec-silvair committed:
Fixed bug with start advertising when already provisioned.

adamgralec-silvair committed:
Added checking if UUID is valid when creating new node.

jakubwitowski-silvair committed:
Add opcode parsing during MessageReceived signal emitting

adamgralec-silvair committed:
Return empty keys on DBus API properties when key not exist.

adamgralec-silvair committed:
Fixed bug when replying for start advertising DBus call.

adamgralec-silvair committed:
Fixed bug with wrong elementy property values.

adamgralec-silvair committed:
Added binding between DBus API and appkey module to provide appkeys getter as DBus property.

adamgralec-silvair committed:
Fixed getting corrupted appkey for DBus property.

jakubwitowski-silvair committed:
Read appkey from storage added

jakubwitowski-silvair committed:
checkpath.pl check

jakubwitowski-silvair committed:
Read dev_key from Json on startup

jakubwitowski-silvair committed:
Fix app_idx and net_idx format problem

jakubwitowski-silvair committed:
Store TTL in Json file when it has been changed instead of on program exit

jakubwitowski-silvair committed:
Change selected stored values from string hex to int

jakubwitowski-silvair committed:
Search for storage directory fixed

michallowasrzechonek-silvair commented on mesh/appkey.c:
```diff
@@ -35,14 +35,15 @@
 #include "mesh/model.h"
 #include "mesh/storage.h"
 #include "mesh/appkey.h"
+#include <stdio.h>
```
remove

michallowasrzechonek-silvair commented on mesh/node.c:
```diff
@@ -1696,25 +1797,61 @@ static bool node_application_keys_getter(struct l_dbus *dbus,
+	if (app_keys_count == 0)
+		goto failed;
+
+	app_keys = (uint8_t **) l_new(uint8_t*, app_keys_count);
```
As far as I remember, app key indexes don't need to be consecutive, so we should rather return a dict, instead or re-indexing keys.

Also, take a look at `l_queue_foreach`

michallowasrzechonek-silvair commented on mesh/node.c:
```diff
@@ -1696,25 +1797,61 @@ static bool node_application_keys_getter(struct l_dbus *dbus,
+	goto flush;
+
+failed:
+	app_keys_count = 1;
```
Why return one key, if there are none defined?

michallowasrzechonek-silvair commented on mesh/node.c:
```diff
@@ -1750,7 +1887,7 @@ static bool node_elements_getter(struct l_dbus *dbus,
 
 		if (!l_dbus_message_builder_append_basic(builder, 'y',
-				&(element->location)))
+				&(element->idx)))
```
I think the value of this dict should be a pair of (location, models), otherwise you can't query element's location at all.

michallowasrzechonek-silvair commented on mesh/appkey.c:
```diff
@@ -270,6 +271,29 @@ static bool set_key(struct mesh_app_key *key, uint16_t app_idx,
 	return true;
 }
 
+bool appkey_get_key_value(struct mesh_app_key *app_key, struct mesh_net *net,
```
Why does an application key value depends on network key refresh? :confused:

michallowasrzechonek-silvair commented on mesh/dbus.c:
```diff
@@ -167,6 +167,23 @@ bool dbus_append_byte_array(struct l_dbus_message_builder *builder,
 	return true;
 }
 
+bool dbus_append_byte_array2d(struct l_dbus_message_builder *builder,
```
I'm not a fan of this function. If anything, this should allow building arrays from `l_queue`. But I'm not sure if this is needed at all. See below.

michallowasrzechonek-silvair commented on mesh/mesh-db.c:
```diff
@@ -502,9 +511,13 @@ bool mesh_db_net_key_add(json_object *jobj, uint16_t idx,
 
-	json_object_object_add(jentry, "keyRefresh",
-					json_object_new_int(phase));
+	if (!jvalue)
```
You probably want to check this before calling `json_object_object_add` ;)

michallowasrzechonek-silvair commented on mesh/mesh-db.c:
```diff
@@ -594,19 +606,20 @@ bool mesh_db_app_key_add(json_object *jobj, uint16_t net_idx, uint16_t app_idx,
-		snprintf(buf, 5, "%4.4x", net_idx);
-		jstring = json_object_new_string(buf);
-		if (!jstring)
+		if (!jvalue)
```
You probably want to check this before calling json_object_object_add ;)

michallowasrzechonek-silvair commented on mesh/mesh-db.c:
```diff
@@ -594,19 +606,20 @@ bool mesh_db_app_key_add(json_object *jobj, uint16_t net_idx, uint16_t app_idx,
+		jvalue = json_object_new_int(net_idx);
+		json_object_object_add(jentry, "boundNetKey", jvalue);
+
+		if (!jvalue)
```
You probably want to check this before calling json_object_object_add ;)

michallowasrzechonek-silvair commented on mesh/mesh-db.c:
```diff
@@ -1275,6 +1288,17 @@ bool mesh_db_read_node(json_object *jnode, mesh_db_node_cb cb, void *user_data)
+	json_object_object_get_ex(jnode, "unicastAddress", &jvalue);
+
+	if (jvalue) {
+		char *str = (char *)json_object_get_string(jvalue);
```
Why casting? Also, `const char *`

michallowasrzechonek-silvair committed:
Fixed empty property values in GetManagedObject.

Instead of using get_uuid_from_path, we're now keeping pointer to our
mesh_node in interface's user_data.

jakubwitowski-silvair committed:
mesh_crypto_aes_ccm_encrypt() function rewrite with ell library

jakubwitowski-silvair committed:
mesh_crypto_aes_ccm_decrypt() function rewrite with ell library

adamgralec-silvair committed:
Made ApplicationKeys (node property on DBus) a dictionary, not array.

jakubwitowski-silvair committed:
Update Makefile.am with cipher lib

jakubwitowski-silvair committed:
Remove storage file and its backup on delete node

jakubwitowski-silvair committed:
Revert "mesh_crypto_aes_ccm_decrypt() function rewrite with ell library"

This reverts commit 0492dbfe82e12c53afeda21925ccf0c4ba0fbe19.

michallowasrzechonek-silvair committed:
Fixed compilation under --enable-maintainer-mode

adamgralec-silvair committed:
Refactored appkey_get_key_info function.

jakubwitowski-silvair committed:
Cleaning after checpath.pl

adamgralec-silvair committed:
Improved AppKey getters flow.

michallowasrzechonek-silvair committed:
Removed unused variable

michallowasrzechonek-silvair committed:
Refactored network key storage

jakubwitowski-silvair committed:
Refactored application key storage

michallowasrzechonek-silvair committed:
Removed unused function

michallowasrzechonek-silvair committed:
Align net_key and app_key implementation

jakubwitowski-silvair committed:
Fix after checpath.pl

jakubwitowski-silvair committed:
net key delete related compilation error fix
